src/components/Modal/index.tsx

```diff
@@ -515,25 +529,34 @@ const Modal: React.FC<ModalProps> = (props) => {
               )}
             </div>
           )}
-          {children && (
-            <div
-              className="modal-body"
-              style={{
-                color: (
-                  isDarkModeOn()
-                    ? 'white'
-                    : undefined
-                ),
-                backgroundColor: (
-                  isDarkModeOn()
-                    ? '#444'
-                    : undefined
-                ),
-              }}
-            >
-              {children}
-            </div>
-          )}
+
+          <div
+            className="modal-body"
+            style={{
+              color: (
+                isDarkModeOn()
+                  ? 'white'
+                  : undefined
+              ),
+              backgroundColor: (
+                isDarkModeOn()
+                  ? '#444'
+                  : undefined
+              ),
+            }}
+          >
+            {isLoading ? (
+              <>
+                <LoadingSpinner />
+                <span className="sr-only">Content loading</span>
```

Comment:
If the programmer wants to create their own loading interface, let them. I think this should only be shown if `children` is undefined, null, an empty array, or falsy in some other way